.github/workflows/pr-build-test/dependencies-zig.ps1

```diff
@@ -0,0 +1,141 @@
+using namespace System.Runtime.InteropServices;
+
+$ErrorActionPreference = "Stop";
+$PSNativeCommandUseErrorActionPreference = $true;
+
+$osArch = [RuntimeInformation]::OSArchitecture
+
+# Determine the host tag
+$zigHost = if ($IsWindows) {
+    switch ($osArch) {
+        "X64" { "x86_64-windows" }
+        "Arm64" { "aarch64-windows" }
+        default { $null }
+    }
+}
+elseif ($IsMacOS) {
+    switch ($osArch) {
+        "X64" { "x86_64-macos" }
+        "Arm64" { "aarch64-macos" }
+        default { $null }
+    }
+}
+elseif ($IsLinux) {
+    switch ($osArch) {
+        "X64" { "x86_64-linux" }
+        "Arm64" { "aarch64-linux" }
+        default { $null }
+    }
+}
+else {
+    $null
+};
+
+if ($null -eq $zigHost) {
+    $osDescription = [RuntimeInformation]::OSDescription;
+    throw "unsupported platform: $osDescription";
+}
+
+# Common variables
+$zigVersion = "0.15.1";
+$zigArchiveFormat = if ($IsWindows) { "zip" } else { "tar.xz" };
+$zigCompilerScriptFormat = if ($IsWindows) { "bat" } else { "sh" };
+$zigBuild = "zig-$zigHost-$zigVersion";
+$zigBuildUrl = "https://ziglang.org/download/$zigVersion/$zigBuild.$zigArchiveFormat";
+$zigHome = Join-Path $HOME "zig";
+$zigBinary = if ($IsWindows) {
+    Join-Path $zigHome $zigBuild "zig.exe"
+} else {
+    Join-Path $zigHome $zigBuild "zig"
+};
+$zigAndRustTargets = @(
+    @{ Zig = "aarch64-linux-gnu"; Rust = "aarch64-unknown-linux-gnu" },
+    @{ Zig = "x86_64-linux-gnu"; Rust = "x86_64-unknown-linux-gnu" }
+    # @{ Zig = "aarch64-windows-gnu"; Rust = "aarch64-pc-windows-gnu" }
+    # @{ Zig = "x86_64-windows-gnu"; Rust = "x86_64-pc-windows-gnu" }
```

Comment:
Do we want to add the windows targets?

Reply:
I've tested whether Zig can completely replace the MinGW toolchain, but without installing MinGW, Zig fails to find libraries like Visual C runtime, so installing MinGW is required. This will be revisited when implementing DSL to configure Zig as the linker.